test/index.ts

```diff
@@ -10,40 +9,73 @@
 // to report the results back to the caller. When the tests are finished, return
 // a possible error to the callback or null if none.
 
-import * as path from 'path';
-import * as Mocha from 'mocha';
-import * as glob from 'glob';
-
-export function run(): Promise<void> {
-	const mocha = new Mocha({
-		ui: 'bdd',
-		useColors: true,
-		timeout: 0
-	});
-
-	const testsRoot = path.resolve(__dirname, '..');
-
-	return new Promise((c, e) => {
-		glob('**/**.test.js', { cwd: testsRoot }, (err, files) => {
-			if (err) {
-				return e(err);
-			}
-
-			// Add files to the test suite
-			files.forEach(f => mocha.addFile(path.resolve(testsRoot, f)));
-
-			try {
-				// Run the mocha test
-				mocha.run(failures => {
-					if (failures > 0) {
-						e(new Error(`${failures} tests failed.`));
-					} else {
-						c();
-					}
-				});
-			} catch (err) {
-				e(err);
-			}
-		});
-	});
+/* eslint-disable eslint-comments/disable-enable-pair */
+/* eslint-disable @typescript-eslint/no-this-alias */
+/* eslint-disable prefer-const */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-var-requires */
+
+"use strict";
+
+import * as glob from "glob";
+import * as paths from "path";
+import * as Mocha from "mocha";
+
+// The test coverage approach is inspired by https://github.com/microsoft/vscode-js-debug/blob/master/src/test/testRunner.ts
+function setupCoverage() {
+    const NYC = require("nyc");
+    const nyc = new NYC({
```

Comment:
nit: I think you can add this config on package.json for `nyc` right? Any reason why it has to be maintained here instead of the global package.json file?

Reply:
This is due to vscode test limitation. I've tried using the command line way to set up nyc such as CLI. However, the generated output looks like the following
```
------------|----------|----------|----------|----------|-------------------|
File        |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
------------|----------|----------|----------|----------|-------------------|
All files   |    72.73 |      100 |      100 |    72.73 |                   |
 runTest.ts |    72.73 |      100 |      100 |    72.73 |          25,26,27 |
------------|----------|----------|----------|----------|-------------------|
```
It seems the vscode-test apis doesn't export source for nyc to use.